internal/s3_agent/s3_agent.go

```diff
@@ -0,0 +1,76 @@
+package s3_agent
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+// S3Agent wraps the s3.S3 structure to allow for wrapper methods
+type S3Agent struct {
+	Client *s3.S3
+}
+
+func NewS3Agent(accessKey, secretKey, endpoint string, debug bool) (*S3Agent, error) {
+	// TODO: cephRegion must not be hardcoded.
+	const cephRegion = "us-east-1"
```

Comment:
@therealak12 
I need your opinion on this line, too. The cephRegion is hardcoded and it shouldn't be. 
Shall I get it from an environment variable?

Reply:
Ok, Sure.